sdk/metricsadvisor/ai-metrics-advisor/src/transforms.ts

```diff
@@ -339,10 +339,14 @@ export function toServiceGranularity(
 }
 
 export function fromServiceDataFeedDetailUnion(original: ServiceDataFeedDetailUnion): DataFeed {
+  const metricMap = new Map();
+  original.metrics.forEach((x) => {
+    metricMap.set(x.name, x.id);
+  });
```

Comment:
How about something like
```suggestion
  const metricMap = new Map(original.metrics.map((x) => [x.name, x.id]));
```

Reply:
included suggestion in latest commit